### PR TITLE
Remove test_api dependency from mockito 4X

### PR DESCRIPTION
**Pull request by jonahwilliams**

### CODE REVIEW

**Comment by jonahwilliams** [comment]
@srawlins

**Comment by srawlins** [comment]
Did you check what the new output is vs the old? Would you mind pasting it?

File: lib/src/mock.dart [hunk 1]
```diff
@@ -857,17 +816,49 @@ class VerificationResult {
...
   }
 }
 
+String _formatFailure(Matcher expected, actual, String which, {String reason}) {
+  var buffer = StringBuffer();
+  buffer.writeln(_indent(_prettyPrint(expected), first: 'Expected: '));
+  buffer.writeln(_indent(_prettyPrint(actual), first: '  Actual: '));
+  if (which.isNotEmpty) buffer.writeln(_indent(which, first: '   Which: '));
+  if (reason != null) buffer.writeln(reason);
+  return buffer.toString();
```
Comment (srawlins) [inline lib/src/mock.dart:842]: This all looks great. I suspect its a very friendly output.

**Comment by jonahwilliams** [comment]
This should be more or less identical, since the code is basically inlined from test api.

**Comment by jonahwilliams** [comment]
let me grab some examples

**Comment by jonahwilliams** [comment]
example test:

```
import 'package:mockito/mockito.dart';
import 'package:test/test.dart';

void main() {
  test('a', () {
    var foo = MockFoo();

    foo.bar();

    verifyNever(foo.bar());
  });

  test('b', () {
    var foo = MockFoo();

    foo.bar();

    verify(foo.bar()).called(2);
  });

  test('c', () {
    var foo = MockFoo();

    foo.fizz('s');

    verify(foo.fizz('d')).called(1);
  });
}

class Foo {
  void bar() {}

  bool fizz(String value) {}
}

class MockFoo extends Mock implements Foo {}

```

with this change:


```
PS C:\Users\Jonah\Documents\mockito> dart test .\test\example_test.dart
00:00 +0 -1: a [E]
  Unexpected calls. All calls: MockFoo.bar()
  package:mockito/src/mock.dart 1098:31  _fail
  package:mockito/src/mock.dart 650:7    _VerifyCall._checkWith
  package:mockito/src/mock.dart 926:18   _makeVerify.<fn>
  test\example_test.dart 11:16           main.<fn>

00:00 +0 -2: b [E]
  Expected: <2>
    Actual: <1>
  Unexpected number of calls

  package:mockito/src/mock.dart 1098:31  _fail
  package:mockito/src/mock.dart 831:5    VerificationResult.called
  test\example_test.dart 19:23           main.<fn>

00:01 +0 -3: c [E]
  No matching calls. All calls: MockFoo.fizz(s)
  (If you called `verify(...).called(0);`, please instead use `verifyNever(...);`.)
  package:mockito/src/mock.dart 1098:31  _fail
  package:mockito/src/mock.dart 644:7    _VerifyCall._checkWith
  package:mockito/src/mock.dart 926:18   _makeVerify.<fn>
  test\example_test.dart 27:11           main.<fn>

00:01 +0 -3: Some tests failed.
```

At 4.x ToT:

```
PS C:\Users\Jonah\Documents\mockito> dart test .\test\example_test.dart
Precompiling executable...
Precompiled test:test.
00:00 +0 -1: a [E]
  Unexpected calls: MockFoo.bar()
  package:test_api                      fail
  package:mockito/src/mock.dart 678:7   _VerifyCall._checkWith
  package:mockito/src/mock.dart 954:18  _makeVerify.<fn>
  test\example_test.dart 11:16          main.<fn>

00:01 +0 -2: b [E]
  Expected: <2>
    Actual: <1>
  Unexpected number of calls

  package:test_api                     expect
  package:mockito/src/mock.dart 875:5  VerificationResult.called
  test\example_test.dart 19:23         main.<fn>

00:01 +0 -3: c [E]
  No matching calls. All calls: MockFoo.fizz(s)
  (If you called `verify(...).called(0);`, please instead use `verifyNever(...);`.)
  package:test_api                      fail
  package:mockito/src/mock.dart 672:7   _VerifyCall._checkWith
  package:mockito/src/mock.dart 954:18  _makeVerify.<fn>
  test\example_test.dart 27:11          main.<fn>

00:01 +0 -3: Some tests failed.
```

**Comment by srawlins** [comment]
Haha that is fantastic. Looks great.

**Pull request merged**